src/main/java/cohen/wordle/WordleController.java

```diff
@@ -0,0 +1,115 @@
+package cohen.wordle;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class WordleController
+{
+    private final WordleGame wordleGame;
+    private final JLabel[][] letters;
+    StringBuilder currentGuess = new StringBuilder(5);
+    private int charsTyped = 0;
+    private int rowCount = 0;
+
+    public WordleController(WordleGame wordleGame, JLabel[][] letters)
+    {
+        this.wordleGame = wordleGame;
+        this.letters = letters;
+    }
+
+    public void addLetter(String letter)
+    {
+        if (charsTyped < 5 && !wordleGame.gameIsWon())
+        {
+            charsTyped++;
+            letters[rowCount][charsTyped - 1].setText(letter.toUpperCase());
+            currentGuess.append(letter.toUpperCase());
+        }
+    }
+
+    public void enterGuess()
+    {
+        if (!currentGuess.toString().equals("") && currentGuess.length() == 5)
```

Comment:
if `currentGuess.length() == 5` then by definition `!currentGuess.toString().equals("")`

You only need `currentGuess.length() == 5`